modules/core/src/main/java/org/apache/ignite/internal/client/thin/ReliableChannel.java

```diff
@@ -118,72 +116,49 @@
      */
     public <T> T service(
         ClientOperation op,
-        Consumer<BinaryOutputStream> payloadWriter,
-        Function<BinaryInputStream, T> payloadReader
+        Consumer<PayloadOutputStream> payloadWriter,
```

Comment:
I do not fully understand an introduction of _payload_ streams. An abstraction of `PayloadOutputStream` seem only enriches `BinaryOutputStream` with a reference to a channel which is only used as an _attachment_ without adding any related behavior. I can imagine alternative approaches:
1. Use a composition, e.g. `ClientResponse` containing `BinaryInputStream` and `ClientChannel`.
2. Use `BiConsumer` and `BiFuctiontion` in `ReliableChannel.service` receiving a channel (or some complex _context_ object) as a second argument.

Reply:
Yes, it was introduced only for a reference to a channel. I've tried to minimize changes to current calls of `ReliableChannel#service`, `ReliableChannel#request`.

Reply:
Do you consider a separate refactoring ticket for improving used abstraction?

Reply:
No, I will fix it in this ticket.